### ISSUE TITLE
Build training pipeline (AdamW + cosine LR + AMP + MixUp)

### ISSUE THREAD

**Issue by loveyadav1015** [loveyadav1015/mini-Swin#2]
## Files to implement
 - [ ] losses.py — label-smoothing cross-entropy; MixUp + CutMix augmentation
 - [ ] scheduler.py — cosine decay with linear warmup (warmup_epochs, min_lr)
 - [ ] trainer.py — Trainer class: forward pass, backward pass, AMP scaler, grad clipping, checkpoint saving
 - [ ] train.py — CLI entry point, loads config YAML, wires everything together, logs to W&B

### Key hyperparameters (from configs/swin_tiny.yaml)

 - Optimizer: AdamW, lr=1e-3, weight_decay=0.05
 - Scheduler: cosine with 20 warmup epochs, min_lr=5e-6
 - AMP: enabled (torch.cuda.amp)
 - Grad clip: 5.0
 - Label smoothing: 0.1

## Acceptance criteria
 - python backend/training/train.py --config configs/swin_tiny.yaml --dataset cifar10 runs without error
 - Loss decreases across first 5 epochs on CIFAR-10
 - Checkpoint saved to backend/checkpoints/ at best validation accuracy
